src/containers/Nodes/Nodes.tsx

```diff
@@ -0,0 +1,193 @@
+import {useCallback, useEffect} from 'react';
+import cn from 'bem-cn-lite';
+import {useDispatch} from 'react-redux';
+
+import DataTable from '@yandex-cloud/react-data-table';
+
+import {AccessDenied} from '../../components/Errors/403';
+import {Illustration} from '../../components/Illustration';
+import {Loader} from '../../components/Loader';
+
+import {Search} from '../../components/Search';
+import {ProblemFilter} from '../../components/ProblemFilter';
+import {UptimeFilter} from '../../components/UptimeFIlter';
+import {EntitiesCount} from '../../components/EntitiesCount';
+
+import routes, {CLUSTER_PAGES, createHref} from '../../routes';
+
+import {
+    ALL,
+    DEFAULT_TABLE_SETTINGS,
+    USE_NODES_ENDPOINT_IN_DIAGNOSTICS_KEY,
+} from '../../utils/constants';
+import {useAutofetcher, useTypedSelector} from '../../utils/hooks';
+import {NodesUptimeFilterValues} from '../../utils/nodes';
+
+import {setHeader} from '../../store/reducers/header';
+import {
+    getNodes,
+    getFilteredPreparedNodesList,
+    setNodesUptimeFilter,
+    setSearchValue,
+    resetNodesState,
+    getComputeNodes,
+} from '../../store/reducers/nodes';
+import {changeFilter, getSettingValue} from '../../store/reducers/settings';
+import {hideTooltip, showTooltip} from '../../store/reducers/tooltip';
+
+import {getNodesColumns} from './getNodesColumns';
+
+import './Nodes.scss';
+
+import i18n from './i18n';
+
+const b = cn('ydb-nodes');
+
+interface NodesProps {
+    tenantPath?: string;
+    className?: string;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    additionalNodesInfo: any;
+}
+
+export const Nodes = ({tenantPath, className, additionalNodesInfo = {}}: NodesProps) => {
+    const dispatch = useDispatch();
+
+    const isClusterNodes = !tenantPath;
+
+    // Since Nodes component is used in several places,
+    // we need to reset filters, searchValue and loading state
+    // in nodes reducer when path changes
+    useEffect(() => {
+        dispatch(resetNodesState());
+    }, [dispatch, tenantPath]);
+
+    const {wasLoaded, loading, error, nodesUptimeFilter, searchValue, totalNodes} =
+        useTypedSelector((state) => state.nodes);
+    const problemFilter = useTypedSelector((state) => state.settings.problemFilter);
+    const {autorefresh} = useTypedSelector((state) => state.schema);
+
+    const nodes = useTypedSelector(getFilteredPreparedNodesList);
+
+    const useNodesEndpoint = useTypedSelector((state) =>
+        getSettingValue(state, USE_NODES_ENDPOINT_IN_DIAGNOSTICS_KEY),
+    );
+
+    const fetchNodes = useCallback(() => {
+        if (tenantPath && !JSON.parse(useNodesEndpoint)) {
+            dispatch(getComputeNodes(tenantPath));
```

Comment:
On older cluster versions getNodes doesn't work as intended in Diagnostics. To fix it by default nodes requested from /compute endpoint if tenantPath is present. 

It can be altered by `USE_NODES_ENDPOINT_IN_DIAGNOSTICS` flag. It is made primarily for developers, to be able to easily check whether clusters versions are update and interface is ready to migrate completely. It can be triggered inside Experiments in Settings